proposals/nnnn-multiple-modules.md

```diff
@@ -0,0 +1,663 @@
+      <description>The module type that the app requests to control.</description>
+    </param>
+    <param name="moduleId" type="String" mandatory="false">
+      <description>Id of a module, published by System Capability. <description>
+   </param>
+    <param name="userLocation" type="Grid" mandatory="false">
+      <description>Location of the user's seat. <description>
+    </param>
+</function>
+
+<function name="GetInteriorVehicleDataConsent" functionID="GetInteriorVehicleDataConsentID" messagetype="response" since="5.x">
+    <param name="allowed" type="Boolean" mandatory="true">
+        <description>"true" - if SDL grants the permission for the requested module;
+        "false" - SDL denies the permission for the requested module.</description>
+    </param>
+    <param name="resultCode" type="Result" platform="documentation" mandatory="true">
+        <description>See Result</description>
+        <element name="SUCCESS"/>
+        <element name="INVALID_DATA"/>
+        <element name="OUT_OF_MEMORY"/>
+        <element name="TOO_MANY_PENDING_REQUESTS"/>
+        <element name="APPLICATION_NOT_REGISTERED"/>
+        <element name="GENERIC_ERROR"/>
+        <element name="REJECTED"/>
+        <element name="IGNORED"/>
+        <element name="DISALLOWED"/>
+        <element name="USER_DISALLOWED"/>
+        <element name="UNSUPPORTED_RESOURCE"/>
+    </param>
+    <param name="info" type="String" maxlength="1000" mandatory="false">
+    </param>
+    <param name="success" type="Boolean" platform="documentation" mandatory="true">
+        <description> true if successful; false, if failed </description>
+    </param>
+</function>
+```
+
+An alternative solution is to add a new RPC `SetUserLocation`, which is sent from mobile apps to SDL and forwarded to HMI, instead of including `userLocation` as a parameter of the `GetInteriorVehicleDataConsent` RPC.
+
+
+### How to use the new parameter for permission control
+
+A simple Module Id based permission control is listed here.
+
+- SDL shall enforce permission control. It may or may not use the local policy table depending on the implementation. The permissions are given by the user/driver via HMI, and information is passed to sdl core.
+
+- When a user launches a RC app from a phone, the app calls GetSystemCapabilites to get vehicle configurations, which includes how many rows, columns and levels are available, how many / what type / what id of remote control modules are available and where are those modules located or designed for. With this info, the app may show a map to the user so that user knows where are the control modules of each type, where the user sits. The app may ask the user to choose user's seat location (at least driver or passenger for dismissal of the lock screen) for later use. The app may get this information from other means, for example scan a RFID tag, a NFC chip or a QrCode. The user can choose a RC module or modules for remote control. 
```

Comment:
```suggestion
- When a user launches a RC app from a phone, the app calls GetSystemCapabilites to get vehicle configurations, which includes how many rows, columns and levels are available, how many / what type / what id of remote control modules are available, and where those modules are located and/or designed for. With this info, the app may show a map to the user so that the user knows where the control modules of each type are, and where the user sits. The app may ask the user to choose their seat location (at least driver or passenger for dismissal of the lock screen) for later use. The app may get this information from other means, for example scan an RFID tag, an NFC chip, or a QrCode. The user can choose a RC module or modules for remote control. 
```

Reply:
updated